bitrise.yml

```diff
@@ -0,0 +1,512 @@
+    - xcode-test@1.18.14:
+        inputs:
+        - project_path: "$BITRISE_PROJECT_PATH"
+        - scheme: "$BITRISE_SCHEME"
+    - xcode-archive@2.4.8:
+        inputs:
+        - project_path: "$BITRISE_PROJECT_PATH"
+        - scheme: "$BITRISE_SCHEME"
+        - export_method: "$BITRISE_EXPORT_METHOD"
+    - deploy-to-bitrise-io@1.3.12: {}
+    - cache-push@2.0.5: {}
+    meta:
+      bitrise.io:
+        stack: osx-xcode-12.2.x
+  primary:
+    steps:
+    - activate-ssh-key@4:
+        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
+    - git-clone@4: {}
+    - cache-pull@2: {}
+    - certificate-and-profile-installer@1: {}
+    - script@1:
+        title: Workaround carthage lipo bug
+        inputs:
+        - content: "#!/usr/bin/env bash\n# fail if any commands fails\nset -e\n# debug
+            log\nset -x\n\necho 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64=arm64
+            arm64e armv7 armv7s armv6 armv8' > /tmp/tmp.xcconfig\necho 'EXCLUDED_ARCHS=$(inherited)
+            $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT))'
+            >> /tmp/tmp.xcconfig\necho 'IPHONEOS_DEPLOYMENT_TARGET=11.4' >> /tmp/tmp.xcconfig\necho
+            'SWIFT_TREAT_WARNINGS_AS_ERRORS=NO' >> /tmp/tmp.xcconfig\necho 'GCC_TREAT_WARNINGS_AS_ERRORS=NO'
+            >> /tmp/tmp.xcconfig\nexport XCODE_XCCONFIG_FILE=/tmp/tmp.xcconfig\nenvman
+            add --key XCODE_XCCONFIG_FILE --value /tmp/tmp.xcconfig "
+    - carthage@3.2:
+        inputs:
+        - carthage_options: " --platform ios --cache-builds"
+        - carthage_command: bootstrap
+    - script@1:
+        inputs:
+        - content: "#!/usr/bin/env bash\n# fail if any commands fails\nset -e\n# debug
+            log\nset -x\n\n\nrm /tmp/tmp.xcconfig \nenvman add --key XCODE_XCCONFIG_FILE
+            --value ''"
+        title: remove carthage lipo workaround
+    - script@1:
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            # fail if any commands fails
+            set -e
+            # debug log
+            set -x
+
+            cd content-blocker-lib-ios/ContentBlockerGen && swift run
+        title: NPM, ContentBlockerGen
+    - cache-push@2:
+        inputs:
+        - cache_paths: |
+            $BITRISE_CACHE_DIR
+            ./Carthage -> ./Carthage/Cachefile
+    - set-xcode-build-number@1:
+        inputs:
+        - plist_path: Blockzilla/Info.plist
+    - set-xcode-build-number@1:
+        inputs:
+        - plist_path: ContentBlocker/Info.plist
+    - set-xcode-build-number@1:
+        inputs:
+        - plist_path: FocusIntentExtension/Info.plist
+    - set-xcode-build-number@1:
+        inputs:
+        - plist_path: OpenInFocus/Info.plist
+    - xcode-archive@2:
+        inputs:
+        - scheme: Focus
+        - export_method: app-store
+    meta:
+      bitrise.io:
+        stack: osx-xcode-12.0.x
+  primary-xcode-10:
```

Comment:
I have not created this one in the past, I think is not used anymore...should we remove it or double check with the person who created it?

Reply:
Do you mean the whole `meta:` section or the fact that this is set to 12.0? I see above it is set to 12.2 and further down I see 12.4 :-)

I would like to upgrade to Xcode 12.5 when I'm done with the Carthage and Glean changes - why don't we leave the workflow as it currently is until that work is that.

Reply:
No, no, I mean the `primary-xcode-10` worfkflow, from L104 to L141 as I think that workflow is not used. Yeah for now the stack should be 12.4, newer workflows I created are using that one.

Context for that is if we don't remove old or unused workflows this file gets really big :)

Reply:
Ah I see. If the workflow is not used, just get rid of it. It it makes sense to keep record of it in Git then I would suggest to commit this PR with those workflows included and then do a followup to remove them. Then we have a record of it in the project history. If that useful.

Reply:
That's a good idea, let's do that and then clean the file as we see things that can be removed as we go. Thanks!